### PR TITLE
Introduce rootkey storage from bakeryv2 upstream

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

More progress on moving off bakeryv2-unstable.
Some tests are updated to use bakeryv2 and the mongo backed root key storage is also ported across.

## QA steps

bootstrap a controller
clear local cookies
login

### CODE REVIEW

File: api/watcher/watcher_test.go [hunk 1]
```diff
@@ -290,24 +291,25 @@ func (s *watcherSuite) assertSetupRelationStatusWatch(
 	// Create a macaroon for authorisation.
 	store, err := s.State.NewBakeryStorage()
 	c.Assert(err, jc.ErrorIsNil)
-	bakery, err := bakery.NewService(bakery.NewServiceParams{
-		Location: "juju model " + s.State.ModelUUID(),
-		Store:    store,
+	b := bakery.New(bakery.BakeryParams{
```
Comment (tlm) [inline api/watcher/watcher_test.go]: (Suggestion), give the b object a better name.

File: featuretests/bakerystorage_test.go [hunk 2]
```diff
@@ -87,30 +85,36 @@ func (s *BakeryStorageSuite) ensureIndexes(c *gc.C) {
...
+	mac, err := s.bakery.Oven.NewMacaroon(context.TODO(), bakery.LatestVersion, nil, bakery.NoOp)
 	c.Assert(err, jc.ErrorIsNil)
 
 	// The background thread that removes records runs every 60s.
 	// Give a little bit of leeway for loaded systems.
 	for i := 0; i < 90; i++ {
-		_, err = s.service.CheckAny([]macaroon.Slice{{mac}}, nil, nil)
+		_, _, err = s.bakery.Oven.VerifyMacaroon(context.TODO(), macaroon.Slice{mac.M()})
 		if err == nil {
 			time.Sleep(time.Second)
```
Comment (ycliuhw) [inline featuretests/bakerystorage_test.go]: not related to current pr, but not sure if we can inject clock and mock for testing.

Comment (wallyworld) [reply featuretests/bakerystorage_test.go]: sadly the clock is hard wired in the upstream repo

File: api/watcher/watcher_test.go (hunk 1 above)
Comment (wallyworld) [reply api/watcher/watcher_test.go]: It's only used a couple of lines below and "bakery" clashes with the package name. "bkry" or something seems worse?

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
[migrate macaroon key storage to v2](https://trello.com/c/bKc2Psub/1017-migrate-macaroon-key-storage-to-v2)

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**